core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -448,4 +475,46 @@ public static MetricsReporter loadMetricsReporter(Map<String, String> properties
 
     return reporter;
   }
+
+  static class CompositeMetricsReporter implements MetricsReporter {
```

Comment:
@rdblue as we discussed I've made this an inner class. However, we could consider moving the class out of `CatalogUtil` and replace https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/TableScanContext.java#L359-L366 with a `CompositeMetricsReporter`.

For cases where we'd like to use multiple reporters during scans
```
TableScan tableScan = table
        .newScan()
        .metricsReporter(...)
        .metricsReporter(...)
        .metricsReporter(...);
```
we would internally instantiate a `CompositeMetricsReporter` in `TableScanContext#reportWith(..)` and then have `TableScanContext#metricsReporter()` return `MetricsReporter` rather than `Collection<MetricsReporter>`.